gremlin-go/driver/traversal.go

```diff
@@ -83,13 +83,13 @@ func (t *Traversal) Iterate() (*Traversal, <-chan bool, error) {
 		return nil, nil, err
 	}
 
-	r := make(chan bool)
+	r := make(chan error)
 	go func() {
 		defer close(r)
 
 		// Force waiting until complete.
-		_, _ = res.All()
-		r <- true
+		_, err = res.All()
```

Comment:
For testing, or for checking in general, would we need to assert these error items in the channel are all nil?

Reply:
Yeah we should actually. Perhaps add assert.NotNil(<-promise) to other test cases that use Iterate().

Reply:
added in https://github.com/lyndonbauto/tinkerpop/pull/55/commits/5a1d2690caf649aff5e13240f86199a708006ffe